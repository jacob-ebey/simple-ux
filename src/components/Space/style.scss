@import '../variables';


.spacer {
  display: flex;
  justify-content: space-between;

  &-even {
    justify-content: space-around;
  }

  &-break-desktop {
    flex-direction: column;
    @include breakpoint(desktop) {
      flex-direction: row;
    }
  }

  &-break-laptop {
    flex-direction: column;
    @include breakpoint(laptop) {
      flex-direction: row;
    }
  }

  &-break-tablet {
    flex-direction: column;
    @include breakpoint(tablet) {
      flex-direction: row;
    }
  }

  &-break-phablet {
    flex-direction: column;
    @include breakpoint(phablet) {
      flex-direction: row;
    }
  }

  &-break-mobileonly {
    flex-direction: column;
    @include breakpoint(mobileonly) {
      flex-direction: row;
    }
  }
}