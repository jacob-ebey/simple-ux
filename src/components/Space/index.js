import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge, mod } from '../tools'

import './style.scss'

const Space = ({ breakpoint, even, class: className, children, ...rest }) => (
  <div
    {...rest}
    class={
      merge(
        'spacer',
        mod('spacer', even && 'even'),
        mod('spacer-break', breakpoint && breakpoint),
        className
      )
    }
  >
    {children}
  </div>
)

export default Space
