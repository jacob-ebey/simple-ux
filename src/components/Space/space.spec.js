/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Space from './'

describe('Space', () => {
  describe('- snapshots -', () => {
    it('can render', () => {
      const wrapper = shallow(<Space />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can space even', () => {
      const wrapper = shallow(<Space even />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Space class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add breakpoint', () => {
      const wrapper = shallow(<Space breakpoint='phablet' />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class and space even', () => {
      const wrapper = shallow(<Space class='custom' even />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can render children child', () => {
      const wrapper = shallow(
        <Space>
          <div>Child 1</div>
          <div>Child 2</div>
          <div>Child 3</div>
        </Space>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
