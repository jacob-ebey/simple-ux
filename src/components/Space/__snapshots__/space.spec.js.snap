// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space - snapshots - can add breakpoint 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer spacer-break-phablet"></div>

`;

exports[`Space - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer custom"></div>

`;

exports[`Space - snapshots - can add custom class and space even 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer spacer-even custom"></div>

`;

exports[`Space - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer"></div>

`;

exports[`Space - snapshots - can render children child 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer">
  <div>Child 1</div>
  <div>Child 2</div>
  <div>Child 3</div>
</div>

`;

exports[`Space - snapshots - can space even 1`] = `
preact-render-spy (1 nodes)
-------
<div class="spacer spacer-even"></div>

`;
